test/gtest/ucp/test_ucp_am.cc

```diff
@@ -799,6 +800,198 @@ UCS_TEST_P(test_ucp_am_nbx, rx_am_mpools,
+    }
+
+    /// @override
+    virtual ucs_status_t
+    am_data_handler(const void *header, size_t header_length, void *data,
+                    size_t length, const ucp_am_recv_param_t *rx_param)
+    {
+        ucs_status_t status;
+        m_am_received = false;
+        status = test_ucp_am_nbx::am_data_handler(header, header_length, data,
+                                                  length, rx_param);
+        cb_counter++;
+        return status;
+    }
+
+private:
+    size_t req_counter = 0;
+    size_t cb_counter  = 0;
+    std::string m_hdr_copy;
+
+    /// @override
+    ucs_status_ptr_t send_am(const ucp::data_type_desc_t &dt_desc,
+                             unsigned flags = 0, const void *hdr = NULL,
+                             unsigned hdr_length = 0,
+                             const ucp_mem_h memh = NULL)
+    {
+        req_counter++;
+        return test_ucp_am_nbx::send_am(dt_desc, flags, hdr, hdr_length, memh);
+    }
+
+    ucs_status_ptr_t fill_sq(ucp::data_type_desc_t &sdt_desc,
+                             const ucp_mem_h memh = NULL,
+                             size_t header_size = 0ul, unsigned flags = 0,
+                             unsigned data_cb_flags = 0)
+    {
+        const double timeout = 5;
+        ucs_status_ptr_t pending_sptr;
+
+        //Warmup for wireup connection
+        pending_sptr = send_am(sdt_desc, get_send_flag() | flags,
+                               m_hdr_copy.data(), header_size, memh);
+        wait_for_flag(&m_am_received);
+        request_wait(pending_sptr);
+        pending_sptr = NULL;
+
+        ucs_time_t deadline = ucs::get_deadline(timeout);
+        while ((ucs_get_time() < deadline) && (pending_sptr == NULL)) {
+            pending_sptr = send_am(sdt_desc, get_send_flag() | flags,
+                                   m_hdr_copy.data(), header_size, memh);
+        }
+
+        return pending_sptr;
+    }
+};
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, send_short, "ZCOPY_THRESH=-1",
+           "RNDV_THRESH=-1")
+{
+    unsigned flags = 0;
+    flags         |= UCP_AM_SEND_FLAG_COPY_HEADER;
+    check_eager(8, 8, flags);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, bcopy_single, "ZCOPY_THRESH=-1",
+           "RNDV_THRESH=-1")
+{
+    unsigned flags = 0;
+    flags         |= UCP_AM_SEND_FLAG_COPY_HEADER;
+    check_eager(fragment_size() / 2, 16, flags);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, bcopy_multi, "ZCOPY_THRESH=-1",
+           "RNDV_THRESH=-1")
+{
+    unsigned flags = 0;
+    flags         |= UCP_AM_SEND_FLAG_COPY_HEADER;
+    check_eager(fragment_size() * 2, max_am_hdr(), flags);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, zcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=256")
+{
+    unsigned flags = 0;
+    flags         |= UCP_AM_SEND_FLAG_COPY_HEADER;
+    check_eager(64, 64, flags);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_send_copy_header, rndv, "ZCOPY_THRESH=2112",
+           "RNDV_THRESH=2112")
```

Comment:
Setting such specific value (2112) suggests some very specific (HW/OS/protocol spec)-related calculation was involved in getting it.

It confuses the potential reader, as one might think "where does this number come from, and what's its meaning?", wasting time on noticing that 2048+64=2112.

AFAIU we just want the msg+hdr be above RNDV thresh, so can simply set `RNDV_THRESH=1000` - like in many other places in the repo.